tools/kotlin-native-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/model/KonanModel.kt

```diff
@@ -16,17 +16,33 @@
 
 package org.jetbrains.kotlin.gradle.plugin.model
 
+import org.jetbrains.kotlin.gradle.plugin.tasks.Produce
+import org.jetbrains.kotlin.konan.target.KonanTarget
+import java.io.File
 import java.io.Serializable
 
 /**
  * An immutable representation of Konan's project model for gradle tooling API.
  * This model is shared with the client processes such as an IDE.
  */
 interface KonanModel : Serializable {
-    val artifacts: List<KonanArtifact>
+    val konanHome: String
+    val konanVersion: String
+
+    val artifacts: List<KonanModelArtifact>
 }
 
-interface KonanArtifact : Serializable {
+/**
+ * A representation of a single binary produced by Kotlin/Native for a cetrain target.
+ */
+interface KonanModelArtifact : Serializable {
     val name: String
-    val path: String
+    val type: Produce
+    val targetPlatform: KonanTarget
+    val path: File
```

Comment:
`path` sounds a little bit weird here.
If it's really just a "path" then it could be a  `String`...

If you need some stuff from the `File` class, why not naming it `file`?

Reply:
Make sense. Fixed.